File: frontend/src/types/payment.ts
export type PaymentStatus =
  | "PENDING"
  | "PAID"
  | "FAILED"
  | "REFUNDED"
  | "EXPIRED";

export interface Payment {
  id: number;
  order_id: number;
  provider: string;
  provider_payment_id: string | null;
  amount: number;
  currency: string;
  status: PaymentStatus;
  payment_url: string | null;
  paid_at: string | null;
  provider_refund_id: string | null;
  refund_reason: string | null;
  refunded_at: string | null;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/payment.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict

from app.constants.payment import PAYMENT_STATUS


class PaymentCreate(BaseModel):
    order_id: int


class PaymentResponse(BaseModel):
    id: int
    order_id: int
    provider: str
    provider_payment_id: str | None
    amount: Decimal
    currency: str
    status: str
    payment_url: str | None
    paid_at: datetime | None
    provider_refund_id: str | None
    refund_reason: str | None
    refunded_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
